File: frontend/backend/models/message.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Message struct {
	ID         uint           `gorm:"primaryKey" json:"id"`
	SenderID   uint           `gorm:"index;not null" json:"senderId"`
	Sender     *User          `gorm:"foreignKey:SenderID" json:"sender,omitempty"`
	ReceiverID uint           `gorm:"index;not null" json:"receiverId"`
	Receiver   *User          `gorm:"foreignKey:ReceiverID" json:"receiver,omitempty"`
	Content    string         `gorm:"type:text;not null" json:"content"`
	IsRead     bool           `gorm:"default:false" json:"isRead"`
	CreatedAt  time.Time      `json:"createdAt"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"-"`
}

func (Message) TableName() string {
	return "messages"
}

File: frontend/src/lib/types/message.ts
import type { Cabinet } from "./cabinet";
import type { Consultation } from "./consultation";
import type { User, Users } from "./users/";
import type { Admin } from "./users/admin";
import type { Doctor } from "./users/doctor";

export type MessageContent = {
    message?: string;
    attachment: Consultation;
}

export type Message = {
    id:                 number;
    sender:             User<any>;
    receiver:           User<any>;
    createdAt:          Date;
    content:            string;
    isRead:             boolean;
}

